source/common/tcp_proxy/BUILD

```diff
@@ -54,3 +64,57 @@ envoy_cc_library(
         "@envoy_api//envoy/extensions/filters/network/tcp_proxy/v3:pkg_cc_proto",
     ],
 )
+
+envoy_cc_library(
+    name = "upstream_interface",
+    hdrs = [
+        "upstream_interface.h",
+    ],
+    deps = [
+        "//include/envoy/tcp:conn_pool_interface",
+    ],
+)
+
+envoy_cc_library(
+    name = "tcp_proxy_upstreams",
+    srcs = [
+        "upstream.cc",
+    ],
+    hdrs = [
+        "upstream.h",
+    ],
+    deps = [
+        ":upstream_interface",
+        "//include/envoy/access_log:access_log_interface",
```

Comment:
do a quick scan and see if you can clean these up?  Alternately TODO and doing in a follow up when you split them up is fine if that's easier.

Reply:
done